src/components/molecules/AnnouncementMessage/AnnouncementMessage.tsx

```diff
@@ -1,59 +1,107 @@
+
+import { isDefined } from "utils/types";
+
+import { useShowHide } from "hooks/useShowHide";
 
 import { RenderMarkdown } from "components/organisms/RenderMarkdown";
 
+import { LinkButton } from "components/atoms/LinkButton";
+
 import "./AnnouncementMessage.scss";
 
-type AnnouncementMessageProps = {
-  message?: string;
-};
+export interface AnnouncementMessageProps {
+  banner: Banner | undefined;
+  announcementForUser?: boolean;
+}
 
 export const AnnouncementMessage: React.FC<AnnouncementMessageProps> = ({
-  message = "",
+  banner,
+  announcementForUser: isAnnouncementForUser = false,
 }) => {
-  const [isVisible, setVisibility] = useState<boolean>(false);
-  const { isExpanded } = useChatSidebarControls();
-
-  const hideAnnouncement = useCallback(() => {
-    setVisibility(false);
-  }, []);
+  const {
+    isShown: isAnnouncementMessageVisible,
+    show: showAnnouncementMessage,
+    hide: hideAnnouncementMessage,
+  } = useShowHide();
 
   useEffect(() => {
-    if (message) {
-      setVisibility(true);
+    if (isDefined(banner?.content)) {
+      showAnnouncementMessage();
     }
-  }, [message]);
+  }, [banner, showAnnouncementMessage]);
+
+  const isActiveButton =
+    banner?.buttonDisplayText && banner?.buttonUrl && banner?.isActionButton;
+
+  const isAnnouncementCloseable =
+    isAnnouncementForUser && !banner?.isForceFunnel;
+
+  const containerClasses = classNames("AnnouncementMessage__container", {
+    "AnnouncementMessage__container--centered": banner?.isFullScreen,
+    "AnnouncementMessage__container--admin": !isAnnouncementForUser,
+  });
+
+  const announcementMessageClasses = classNames("AnnouncementMessage", {
+    "AnnouncementMessage--fullscreen":
+      banner?.isFullScreen && isAnnouncementForUser,
+    "AnnouncementMessage--user": isAnnouncementForUser,
+  });
 
-  if (!isVisible || !message) return null;
+  const handleBannerModalClose = () => {
+    if (banner?.isForceFunnel) return;
+
+    hideAnnouncementMessage();
+  };
+
+  if (!isAnnouncementForUser && !banner?.content)
+    return (
+      <div className="AnnouncementMessage">
+        <span className="AnnouncementMessage__default-text">
+          No announcement
+        </span>
+      </div>
+    );
+
+  if (!isAnnouncementMessageVisible) return null;
 
   return (
-    <div
-      aria-labelledby="announcement-container-message"
-      role="dialog"
-      className={classNames("announcement-container", {
-        centered: !isExpanded,
-      })}
-    >
-      <div className="announcement-message" id="announcement-container-message">
-        <RenderMarkdown text={message} />
+    <>
+      <div className={containerClasses} onClick={handleBannerModalClose} />
+
+      <div className={announcementMessageClasses}>
+        {banner?.title && (
+          <h2 className="AnnouncementMessage__title">{banner.title}</h2>
+        )}
+
+        <div className="AnnouncementMessage__content">
+          <RenderMarkdown text={banner?.content} />
```

Comment:
```suggestion
          <RenderMarkdown text={banner.content} />
```

Reply:
not gonna work, `banner` is not required upon venue creation, but later we should display it at /venue/admin page regardless

Reply:
but you have `useEffect` which handles not defined state.

```ts
  useEffect(() => {
    if (isDefined(banner?.content)) {
      showAnnouncementMessage();
    }
  }, [banner, showAnnouncementMessage]);
```

and then you do
```ts
  if (!isAnnouncementMessageVisible) return null;
```